src/Scriban/Syntax/Expressions/ScriptPipeCall.cs

```diff
@@ -69,13 +69,20 @@ public override object Evaluate(TemplateContext context)
 
                 return result;
             }
+            catch (ScriptRuntimeException sre) when (sre.OriginalMessage.StartsWith("Invalid target function"))
```

Comment:
It is too brittle to rely on exception message. I would prefer that in `ScriptFunctionCall` we directly detect the pipe and throw an exception with the correct message

You can move the part where you have this test:

```c#
var isPipeCall = processPipeArguments && context.CurrentPipeArguments != null && context.CurrentPipeArguments.Count > 0
```
And use this variable later in the function.